health-services/ExerciseSampleCompose/app/src/main/java/com/example/exercisesamplecompose/presentation/ExerciseSampleApp.kt

```diff
@@ -0,0 +1,119 @@
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.exercisesamplecompose.presentation
+
+import android.content.ContentValues.TAG
+import android.util.Log
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.platform.LocalContext
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.navigation.NavHostController
+import androidx.navigation.NavType
+import androidx.navigation.navArgument
+import androidx.wear.compose.navigation.SwipeDismissableNavHost
+import androidx.wear.compose.navigation.composable
+import com.example.exercisesamplecompose.Screens
+import com.example.exercisesamplecompose.data.HealthServicesManager
+import kotlinx.coroutines.launch
+
+/** Navigation for the exercise app. **/
+
+@Composable
+fun ExerciseSampleApp(
+    healthServicesManager: HealthServicesManager,
+    navController: NavHostController,
+    startDestination: String
+) {
+    val composableScope = rememberCoroutineScope()
+
+    SwipeDismissableNavHost(
+        navController = navController, startDestination = startDestination
+    ) {
+        composable(Screens.StartingUp.route) {
+            val viewModel: ExerciseViewModel = viewModel(
+                factory = ExerciseViewModelFactory(
+                    healthServicesManager = healthServicesManager, LocalContext.current
+                )
+            )
+
+            StartingUp(onAvailable = {
+                navController.navigate(Screens.PreparingExercise.route)
+            }, onUnavailable = {
+                navController.navigate(Screens.ExerciseNotAvailable.route)
+            }, hasCapabilities = viewModel.hasExerciseCapabilities.value
+                //healthServicesManager = healthServicesManager
+            )
+            Log.d(TAG, viewModel.hasExerciseCapabilities.value.toString())
+        }
+        composable(Screens.PreparingExercise.route) {
+            val viewModel: ExerciseViewModel = viewModel(
+                factory = ExerciseViewModelFactory(
+                    healthServicesManager = healthServicesManager, LocalContext.current
+                )
+            )
+            val serviceState by viewModel.exerciseServiceState
+            val permissions = viewModel.permissions
+            PreparingExercise(onStartClick = {
+                viewModel.connectToService()
+                navController.navigate(Screens.ExerciseScreen.route)
+            },
+                prepareExercise = { composableScope.launch { viewModel.prepareExercise() } },
```

Comment:
Does it make more sense to use the viewModelScope in the ViewModel? If the composable exits but the viewModel still survives?

Reply:
Changed!